src/FSharp.Formatting.Common/Menu.fs

```diff
@@ -0,0 +1,55 @@
+module FSharp.Formatting.Common.Menu
+
+open System
+open System.IO
+open FSharp.Formatting.Templating
+
+type MenuItem = { Link: string; Content: string }
+
+let private snakeCase (v: string) =
```

Comment:
This still produces kinda funky identifiers.
Consider "End-user documentation" being transformed to "_end-userdocumentation".
I would trim the leading and trailing `_` and replace spaces with `-` or `_` instead.
And may even replace all `_` to `-` or vice versa, so you don't have a mix.

Feel free to revisit this entire helper function if that makes more sense.